lib/plug/adapters/test/conn.ex

```diff
@@ -31,13 +31,15 @@ defmodule Plug.Adapters.Test.Conn do
         })
     }
 
+    split_on_dot? = Map.has_key?(conn.private, :split_on_dot)
```

Comment:
I don't think this is going to work as you expect beyond testing because Cowboy won't actually split on dot. :) And passing this option (split_on_dot) to Cowboy will definitely break stuff. Imagine you have something like:

    /user/jose.valim

Which should match:

    /user/:username

If you split all Cowboy requests on dot, then this will be suddenly broken. :)

Reply:
Aargh! I see plug_cowboy's [`conn/1`](https://github.com/elixir-plug/plug_cowboy/blob/f82f2ff982f04fb4faa3a12fd2b08a7cc56ebe15/lib/plug/cowboy/conn.ex#L21) is different from the test `conn/4` in plug. It seems not feasible to split `.` only on certain set condition, as the `req` data passed to the function is limited.